mava/utils/checkpointing.py

```diff
@@ -192,7 +193,16 @@ def restore_params(
         # Restore hidden states if required
         restored_hstates = None
         if restore_hstates and THiddenState is not None:
-            restored_hstates = THiddenState(**restored_learner_state_raw["hstates"])
+            if "hidden_state" in restored_learner_state_raw.keys():
+                restored_hstates = THiddenState(
+                    encoder_hstate=restored_learner_state_raw["hidden_state"]["encoder_hstate"],
+                    decoder_hstate=(
+                        restored_learner_state_raw["hidden_state"]["decoder_hstate"][0],
+                        restored_learner_state_raw["hidden_state"]["decoder_hstate"][1],
+                    ),
+                )
+            else:
+                restored_hstates = THiddenState(**restored_learner_state_raw["hstates"])
```

Comment:
Not sure why we need this? Shouldn't `**` unpack correctly?

Reply:
it worked on my side 😅